### PR TITLE
chore(test): upgrade to ava v3.15.0

### PR DESCRIPTION
**Pull request by compwright**
Also replaces deprecated lolex with @sinonjs/fake-timers

### CODE REVIEW

**Comment by compwright** [comment]
@tmeasday

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "A simple, fast, robust job/task queue, backed by Redis.",
   "main": "index.js",
   "dependencies": {
-    "p-finally": "^1.0.0",
+    "p-finally": "^2.0.0",
```
Comment (bradvogel) [inline package.json:7]: Are there any breaking changes in this update? I couldn't find the changelog..

Comment (bradvogel) [reply package.json:7]: Oh, thanks! I missed that.

**Review by bradvogel — APPROVED** [review]

**Pull request merged**